source/constructs/api/data_source/glue_database_detector.py

```diff
@@ -0,0 +1,125 @@
+import os
+
+import boto3
+import logging
+
+from common.constant import const
+from common.enum import ConnectionState, DatabaseType, Provider
+from db.database import get_session
+from db.models_data_source import DetectionHistory, RdsInstanceSource, Account
+from . import crud
+from . import service
+from catalog.service import delete_catalog_by_database_region
+from sqlalchemy.orm import Session
+import asyncio
+
+sts_client = boto3.client('sts')
+admin_account_region = boto3.session.Session().region_name
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+async def detect_glue_database_connection(session: Session, aws_account_id: str):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 21 other functions. By comparison, 99% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this function or break it into multiple functions.